src/shoobx/wfmc/process.py

```diff
@@ -901,8 +920,8 @@ def evaluateInputs(process, formal, actual, evaluator, strict=True):
             __traceback_info__ = (parameter, expr)
             try:
                 value = evaluator.evaluate(expr)
-            except:
-                if strict:
+            except Exception as e:
+                if strict and not isinstance(e, ignored_exceptions):
```

Comment:
`ignored_exceptions` is not a good name. It may seem that it is exceptions that are going  to be ignored while evaluating the expression, but it is the opposite - exceptions that are going to be raised (specifically _not_ ignored).

Also, passing a set of exceptions to a function is a  pretty unorthodox pattern.  I suggest to introduce and catch some very specific `CannotEvaluateException` here and let `evaluator` raise it when necessary. This will make it more flexible and less weird.

Reply:
Actually Exceptions will be ignored if it is found in `ignored_exceptions`. Notice `not` used, so it will behave same way as non strict, but for specific exception.

And with `CannotEvaluateException` you want to just raise this, instead of any caught exception? Or something else? Cause if we just raise this, we will still have same problem, because we need to handle exceptions for each item. If we raise it, iteration stops and no inputs will be returned.

Reply:
> Actually Exceptions will be ignored if it is found in ignored_exceptions. Notice not used, so it will behave same way as non strict, but for specific exception.

I understand how it works, but it is a confusing naming. It works the opposite way than it looks like to a casual reader. To me this snippet means "do something, but if ValueError happens while doing it, ignore it". However in case of this function, it works the opposite way.

```
do_something(ignored_exceptions=(ValueError,))
```

> And with CannotEvaluateException you want to just raise this, instead of any caught exception?

I suggest making it an `IPythonExpressionEvaluator`' responsibility to raise `CannotEvaluateException` if it cannot evaluate the expression given to it. And only expect `CannotEvaluateException` here. Users of `shoobx.wfmc` can override the implementation of `IPythonExpressionEvaluator` (and that's what the only user of this lib is doing anyway) and catch specific exceptions there.